### DISCUSSION TITLE
How to properly use M1

### DISCUSSION THREAD

**Discussion by dannyboid** [Chia-Network/chia-blockchain#4098] [Q&A]
Hello,

Specs :

MacBook Pro M1 2020
Memory 16

1TB Sandisk Extreme Portable SSD
(Up to 550 mb/s^1)

2x of 8TB Seagate Backup Plus Hub

About me:
Just to let you all know, I am very new to this. I don’t know much about scripts/codes my only experience was back in school where we created a very basic game and that was many years ago. Ive been buying/trading cryptos for 2 years now and finally decided to begin mining/farm. Thought chia would be a good place to learn because its relatively cheap (as in I don’t need the most epic computer to farm/mine).

Questions:
What should be my next step?
Is there any tutorials or articles that I can read? Can someone kindly explain? I read a lot of information but they’re all for windows so I’m not sure what I could apply. I did read through a lot of Reddit forums regarding info to M1 but its very difficult for me to follow through because like I said earlier I don’t have much experience in scripts.
A thing that I noticed a lot is using a python program so, I’ve downloaded the Python 3.9.5 from https://www.python.org/downloads/mac-osx/ I have the program on but I didn’t type any script because I don’t know what I’m doing.
 Also I read about running it natively will make it quicker, How do I check if its running natively? Ive downloaded everything from the chia website.

So far I have 2 plots than I ran in the default setting, 1 plot for each HDD. It took me around 64800 seconds (18 hours) to acquire the 2 plots.

All help will be greatly appreciated
Thanks in advance!

**Comment by ghost** [comment]
Hello

I did the following on my M1 Mac Mini:
- Install Python from [python.org](https://www.python.org/downloads/mac-osx/), select the latest universal2 installer
- Install [Rust](https://www.rust-lang.org/learn/get-started)
- Follow the [Chia-Network INSTALL steps](https://github.com/Chia-Network/chia-blockchain/wiki/INSTALL#MacOS) for Mac OS, I did not install the GUI part
-- Open terminal
-- git clone https://github.com/Chia-Network/chia-blockchain.git -b latest
-- cd chia-blockchain
-- sh install.sh
-- . ./activate
- Now you should be able to execute the "chia" command.. if this works without error, then you're all set to start plotting.

I propose you execute a few plots through a terminal session before trying to automate plotting using Automator and launchd.
So go ahead and execute something like:
```
caffeinate -m -i chia plots create \
        --tmp_dir ${plots_tmp_dir}/${plots_id} \
        --final_dir ${plots_final_dir} \
        --size ${plots_size} \
        --num ${plots_num} \
        --num_threads ${plots_num_threads} \
        --farmer_public_key ${plots_farmer_public_key} \
        --pool_public_key ${plots_pool_public_key} \
        ${plots_options} > ${plots_tmp_dir}/${plots_id}.txt
```
I used the above command in a docker container, replace the variables with your own values.
*caffeinate* is used to prevent the Mac going to sleep - sleep would result in failed plots.

I have the following setup:
- Mac Mini M1, 16GB Ram
- 2x Samsung T7 external SSD 1TB

With this setup I am able to plot two tasks in parallel without stressing swap to much:
- size k32 (100GB) in 14'000 seconds / less han 4 hours
- size k33 (200GB) in 43'500 seconds / about 12 hours

### Challenges
1. Automate the script execution so that the performance matches the Interactive session
2. Automate so that the script has the permissions to required Volumes
3. Write a script to upload the finished plots to the farm
4. Learn how to keep the chia-blockchain up to date

### Inspiration
You might want to have a look at my forked [chia-docker repository](https://github.com/gldecurtins/chia-docker), especially into the [scripts directory](https://github.com/gldecurtins/chia-docker/tree/main/scripts).

```
grep "Total time" *.txt
plot-k32-2021-05-09-22-08.txt:Total time = 14809.931 seconds. CPU (126.460%) Mon May 10 02:15:42 2021
plot-k32-2021-05-09-22-08.txt:Total time = 14121.762 seconds. CPU (134.770%) Mon May 10 06:20:44 2021
plot-k32-2021-05-09-22-08.txt:Total time = 14571.670 seconds. CPU (132.510%) Mon May 10 10:29:57 2021
plot-k32-2021-05-10-11-05.txt:Total time = 14257.890 seconds. CPU (134.250%) Mon May 10 15:03:28 2021
plot-k32-2021-05-10-15-13.txt:Total time = 13761.673 seconds. CPU (131.450%) Mon May 10 19:02:56 2021
plot-k32-2021-05-10-19-16.txt:Total time = 13859.481 seconds. CPU (131.640%) Mon May 10 23:07:06 2021
plot-k32-2021-05-10-23-25.txt:Total time = 14287.297 seconds. CPU (132.260%) Tue May 11 03:23:51 2021
plot-k32-2021-05-11-03-35.txt:Total time = 13862.240 seconds. CPU (131.100%) Tue May 11 07:26:56 2021
plot-k32-2021-05-11-07-38.txt:Total time = 13490.689 seconds. CPU (133.180%) Tue May 11 11:23:34 2021
plot-k32-2021-05-11-11-35.txt:Total time = 14183.155 seconds. CPU (131.240%) Tue May 11 15:31:40 2021
```

Happy plotting!
-Luca.

**Reply by dannyboid** [reply]
Wow! thanks for the step by step approach. i'm now able to plot a lot better and learned so much

**Reply by waitzShigoto** [reply]
Thanks for sharing the method that you use, get a lot knowledge from your article.

Could I ask a question about that can this approach plotting several  task at a same time and consume less time?
Or it just can plotting one task and less than 4 hour?

I would like to plot 4 k32 task at same time but I,m not sure if this can afford at mac mini 16gb m1 device?

Thanks for your sharing.

**Reply by ghost** [reply]
Hi, I keep two plot create jobs running in parallel. Each plot create job uses a dedicated external SSD storage. I see the limiting factor is the available RAM - and the heavy usage of the swap.

You could experiment a little and see how many in parallel you can trigger:
- Turn on bitfield - according to the https://www.chia.net/2021/02/22/plotting-basics.html article it could have some minor impact
- Do not start the plotters at the same time, so that they are not in the same phase
- Try to move the swap to an external device, so that you do not harm the M1 Mac

Give it a try. Start 4 of them (with or without --nobitfield) and keep a close eye on the activity monitor, especially the Memory tab. If you notice a constant fluctuation of swap I would consider it unsustainable and start over.
